gnomad/utils/annotations.py

```diff
@@ -1138,3 +1198,219 @@ def hemi_expr(
+            " returned."
+        )
+
+    # Define VRS Attributes that will later be read into the dictionary to be returned.
+    vrs_id = f"{ht.info.vrs.VRS_Allele_IDs[1].collect()[0]}"
+    vrs_chrom_id = f"{chrom_dict[chr_in]}"
+    vrs_start_value = ht.info.vrs.VRS_Starts[1].collect()[0]
+    vrs_end_value = ht.info.vrs.VRS_Ends[1].collect()[0]
+    vrs_state_sequence = f"{ht.info.vrs.VRS_States[1].collect()[0]}"
+
+    # Defining the dictionary for VRS information.
+    vrs_dict = {
+        "_id": vrs_id,
+        "location": {
+            "_id": "to-be-defined",
+            "interval": {
+                "end": {"type": "Number", "value": vrs_end_value},
+                "start": {
+                    "type": "Number",
+                    "value": vrs_start_value,
+                },
+                "type": "SequenceInterval",
+            },
+            "sequence_id": vrs_chrom_id,
+            "type": "SequenceLocation",
+        },
+        "state": {"sequence": vrs_state_sequence, "type": "LiteralSequenceExpression"},
+        "type": "Allele",
+    }
+
+    # Set location ID
+    location_dict = vrs_dict["location"]
+    location_dict.pop("_id")
+    location = ga4gh_vrs.models.SequenceLocation(**location_dict)
+    location_id = ga4gh_core._internal.identifiers.ga4gh_identify(location)
+    vrs_dict["location"]["_id"] = location_id
+
+    logger.info(vrs_dict)
+
+    # If vrs_only was passed, only return the above dict and stop.
+    if vrs_only:
+        return vrs_dict
+
+    # Create a list to then add the dictionaries for frequency reports for
+    # different ancestry groups to.
+    list_of_group_info_dicts = []
+
+    # Define function to return a frequency report dictionary for a given group
+    def _create_group_dicts(
+        variant_ht: hl.Table,
+        group_index: int,
+        group_id: str,
+        group_label: str,
+        vrs_id: str,
+    ) -> dict:
+        """
+        Return a dictionary for the frequency information of a given variant for a given subpopulation.
+
+        :param variant_ht: Hail Table with only one row, only containing the desired variant.
+        :param group_index: Index of frequency within the 'freq' annotation containing the desired group.
+        :param group_id: String containing variant, genetic ancestry group, and sex (if requested). Example: "chr19-41094895-C-T.afr.XX".
+        :param group_label: String containing the full name of genetic ancestry group requested. Example: "African/African American".
+        :param vrs_id: String containing the VRS ID of the variant in ht_subpop.
+        :return: Dictionary containing VRS information (and genetic ancestry group if desired) for specified variant.
+
+        """
+        # Obtain frequency information for the specified variant
+        group_freq = variant_ht.freq[group_index]
+
+        # Dictionary to be returned containing information for a specified group
+        freq_record = {
+            "id": group_id,
```

Comment:
it looks like this id should include the variant as well (https://github.com/ga4gh/va-spec/blob/9a08cd2863536fd1484e55018156901ee771e188/docs/Modeling/Population%20Frequency/simple_result_example.json#L50)

Reply:
and within these records it also looks like the genetic ancestry groups are in all caps

Reply:
changed to "id":f"{variant}.{group_id.upper()}" in coming commi t

Reply:
this chance should resolve both of the comments above this one, if I'm interpreting them right